tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,370 @@
+		vpsrules[name] = d
+	} else {
+		return fmt.Errorf("vps rule: %s is already registered", name)
+	}
+
+	return nil
+}
+
+// 
+func GetVpsRules() map[string] VpsTemplate {
+	return vpsrules
+}
+
+//-------
+type Vpscase1 struct {
+	//Case description
+	name string
+	// Enabled
+	enabled bool
+}
+
+//# Case-1--enforcemnt: Required
+
+func (v * Vpscase1) GetLabels() [] labelDict {
+
+	lbldata := []labelDict{}
+	node1lbl := labelDict{"media_type": "SSD","vps_test":"test"}  
+	node2lbl := labelDict{"media_type": "SATA","vps_test":"test"}
+	node3lbl := labelDict{"media_type": "SSD","vps_test":"test"}
+	node4lbl := labelDict{"media_type": "SSD","vps_test":"test"}
+	lbldata = append(lbldata, node1lbl, node2lbl,node3lbl,node4lbl)
+		return lbldata
+}
+
+func (v * Vpscase1) GetPvcNodeLabels( lblnodes map[string][]string) map[string]map[string][]string {
+
+	for key,val := range lblnodes {
+		logrus.Infof("label node: key:%v Val:%v", key,val)
+	}
+
+	//Create 3 node lists (requiredNodes, prefNodes, notOnNodes)
+	volnodelist := map[string]map[string][]string{}
+	volnodelist["mysql-data"] = map[string][]string{}
+	volnodelist["mysql-data-seq"] = map[string][]string{}
+	volnodelist ["mysql-data"]["pnodes"] = [] string{}
+	volnodelist ["mysql-data"]["nnodes"] = [] string{}
+	volnodelist ["mysql-data-seq"]["pnodes"] = [] string{}
+	volnodelist ["mysql-data-seq"]["nnodes"] = [] string{}
+
+
+	for _,lnode := range lblnodes["media_typeSSD"] {
+		volnodelist ["mysql-data"]["rnodes"] = append(volnodelist ["mysql-data"]["rnodes"], lnode)
+		volnodelist ["mysql-data-seq"]["rnodes"] = append(volnodelist ["mysql-data-seq"]["rnodes"], lnode)
+	}
+/*
+	for _,lnode := range lblnodes["media_typeSATA"] {
+		volnodelist ["mysql-data"]["rnodes"] = append(volnodelist ["mysql-data"]["rnodes"], lnode)
+		volnodelist ["mysql-data-seq"]["rnodes"] = append(volnodelist ["mysql-data-seq"]["rnodes"], lnode)
+	}
+*/
+	return volnodelist
+}
+
+
+/*
+ * 1. Each rule template, will provide the expected output
+ */
+
+func (v * Vpscase1) Validate(appVolumes []*volume.Volume,volscheck map[string]map[string][]string) {
+
+	logrus.Infof("Deployed volumes:%v,  volumes to check for nodes placement %v ",
+			appVolumes, volscheck)
+
+	for _, appvol := range appVolumes {
+
+		for vol, vnodes := range volscheck {
+
+			if appvol.Name == vol {
+				replicas, err := Inst().V.GetReplicaSetNodes(appvol)
+				logrus.Infof("==Replicas for vol: %s, appvol:%v Replicas:%v ", vol, appvol,replicas)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(replicas).NotTo(BeEmpty())
+
+				// Must have (required)
+				for _,mnode := range vnodes["rnodes"]  {
+					found := ""
+					for _,rnode := range replicas {
+						logrus.Infof("Expected Volume Node:%v Replica Node:%v", mnode, rnode)
+						if mnode == rnode {
+							found=rnode
+							break	
+						}
+					}
+					 if found == "" {
+						logrus.Errorf("Volume '%v' does not have replica on node:'%v'", appvol,mnode)
+					  	Expect(found).NotTo(BeEmpty())
```

Comment:
you can add the description in the `NotTo(` itself and remove the logrus above and also the `if fount == ""` check.

The point of Expect is to internally check the condition. So we should not be checking for ` if found == "" {`

Reply:
Do the same for other places in the PR.